src/jshint.js

```diff
@@ -2002,6 +2035,19 @@ var JSHINT = (function() {
     error("E014");
   };
   assignop("%=", "assignmod", 20);
+  assignop("**=", function(left, that) {
+    if (!state.inES7()) {
```

Comment:
The `esversion` option supports both "traditional" values (e.g. 5, 6, 7) and "year-based" values (though it's currently a little broken\*). I think we're internally consistent, though we're using the "traditional" scheme. So method names like `inES7` seem logical enough to me.

\* I've had a pull request open to fix that for a little while now: https://github.com/jshint/jshint/pull/3054. If you get a chance, could you take a look at that, as well?